adagrams/game.py

```diff
@@ -1,13 +1,129 @@
+
+    hand = [] # These represent the hand of letters that the player has drawn
+    random_index = 0
+    for i in range(10):
+        random_index = randint(0, len(letter_pool) - 1) # The letters should be randomly drawn from a pool of letters
+        
+
+        while letter_pool[random_index] == 0: # If the randomly drawn letter has already been drawn, then draw again
+            random_index = randint(0, len(letter_pool) - 1)
+        hand.append(letter_pool[random_index])
+        letter_pool[random_index] = 0 # Once a letter is drawn, it cannot be drawn again until the pool is reset
+
+        #  alternative approach without setting to 0:
+        # hand.append(letter_pool.pop(random_index))
+
+
+    return hand # Returns an array of ten strings
+
+def uses_available_letters(word, letter_bank): # Has two parameters:
+    """
+    Checks if the input word can be formed from the letters in the letter bank.
+    
+    Parameters:
+        word (string): describes some input word.
+        letter_bank (list): describes an array of drawn letters in a hand.
+
+    Returns:
+        type: True of False.
+    """
+    # Returns True if every letter in the input word is available (in the right quantities) in the letter_bank
+    # Returns False if not; if there is a letter in input that is not present in the letter_bank or has too much of compared to the letter_bank
+    list_of_available_letters = []
+    for letter in letter_bank:
+        list_of_available_letters.append(letter.upper())
+    
+    for letter in word:
+        if letter.upper() in list_of_available_letters:
+            
+            # list_of_available_letters.remove(letter.upper())
+            
+            # alternative approach without using remove():
+            for i in range(len(list_of_available_letters)):
+                if list_of_available_letters[i] == letter.upper():
+                    list_of_available_letters[i] = 0
+                    break
+        else:
+            return False
+        
+    return True
 
 def score_word(word):
-    pass
+    """
+    Returns the score of a given word as defined by the Adagrams game.
+    
+    Parameters:
+        word (string): string of characters.
+
+    Returns:
+        int: number of points.
+    """
+    # Each letter's point value is described in the table below
+    score_chart = { "A": 1, "B": 3, "C": 3, "D": 2, "E": 1, "F": 4, "G": 2, "H": 4, "I": 1, "J": 8, "K": 5, "L": 1, "M": 3, "N": 1, "O": 1, "P": 3, "Q": 10, "R": 1, "S": 1, "T": 1, "U": 1, "V": 4, "W": 4, "X": 8, "Y": 4, "Z": 10 }
+    
+    score = 0
+
+    # Each letter within word has a point value. The number of points of each letter is summed up to represent the total score of word
+    for letter in word:
+        if letter.upper() in score_chart:
+            score += score_chart[letter.upper()]
+    if len(word) >= 7:
+                score += 8 # If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
```

Comment:
8 is another magic number. If we use a descriptively named variable, we can make our intent known without leaving a comment.

```suggestion
                BONUS_POINTS_FOR_LENGTH = 8
                score += BONUS_POINTS_FOR_LENGTH
```